Add releases.md

jeddy3 opened the pull request:
Closes #2700

@hudochenkov, @ntwb @evilebottnawi If you’d like access to the npm package for performing releases, please pop your npm usernames below and I’ll added you. Likewise, if you’d like access to the stylelint twitter account you’ll need to drop an email address or something so I can send the access details over.

ntwb commented:
We should make note that you need a GitHub API token

Because of issues I had for a while I was not always reliably exporting my GitHub API token to my shell, so when I ran `npm release` for `stylelint-config-wordpress` it would fail half way through, as such I [changed the npm](https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress/blob/master/package.json#L58) script to `"release": "npmpub --verbose",` for added debugging, this allowed me to then export my GitHub API token and to manually step through the changes that failed in the terminal 😖 

The `npmpub` module also [comes with a](https://www.npmjs.com/package/npmpub#options)  `--dry` option to perform a _dry test run_ that is useful, per the above I occasionally hack that into my `package.json` to confirm everything is right to go 😬 

Do you think it's worth updating the stylelint npm tasks to include the `--verbose` flag, and a new dry run type task like `"release:dryrun": "npmpub --verbose -dry", or just include some of this in the proposed PR docs here?

jeddy3 commented:
> Do you think it's worth updating the stylelint npm tasks to include the --verbose flag, and a new dry run type task like `"release:dryrun": "npmpub --verbose -dry",

If doing so will help catch missing GitHub API tokens, then I think it's worth it. 

I've added a `dry-release` task (as I believe the colon notation is for tasks that are intended to be run in parallel e.g. `lint:*`) and added an additional step to the process in the `.md` file.

I'll add the task to the `stylelint-config-*` repos too.

ntwb commented:
p.s. Happy to be added as a _backup user_ for npm and Twitter if you want another user with access: 

NPM: `netweb` Email: `stephen@netweb.com.au`

hudochenkov commented on docs/developer-guide/releases.md:
```diff
@@ -0,0 +1,45 @@
+
+1.  Locally test `master` in the `stylelint-config-*` shareable configs repos.
+2.  Locally test `master` in the `stylelint.io` repo.
+3.  Locally test `master` in the `stylelint-demo` repo.
```
What does it mean? `npm i stylelint/stylelint && npm test` in every repo?

jeddy3 replied:
Yep

hudochenkov commented on docs/developer-guide/releases.md:
```diff
@@ -0,0 +1,45 @@
+    1.  Run `npm install -S stylelint@latest`
+    2.  Run `npm test`
+    3.  Commit and _push up_ these changes.
+    1.  Go to [https://stylelint.io/demo](https://stylelint.io/demo) and confirm the update was automatically deployed.
```
How to verify that?

jeddy3 replied:
Check that a feature added or bug fixed in the release works now works at https://stylelint.io/demo

jeddy3 commented:
> When do we need a GitHub API token?

You'll get a warning you do the dry run.
